### ISSUE TITLE
Calendar slow render with datatable

### ISSUE THREAD

**Issue by igogu** [primefaces/primeng#2404]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**

when a p-datatable has the p-calendar component in his row, its only needed 5-6 row for making the page slow and the calendar rendering lates like 1 second to respond to the click, 

**Expected behavior**
Instant render after calendar clicked aswell the other components in the page

**Minimal reproduction of the problem with instructions**

    <p-dataTable [value]="listado | async" [globalFilter]="gb" 
    .....">    
        <p-column *ngFor="let col of colus" [field]="col.field"....>

			<template let-row="rowData" let-ri="rowIndex" pTemplate>
				...
				<div *ngIf="col.fieldtype=='date'">
				 	<p-calendar [ngModel]="value | date: 'dd/MM/yyyy'" [locale]="es" dateFormat="dd/mm/yy" [monthNavigator]="true"
				  		[yearNavigator]="true" (ngModelChange)="value=$event">
				  	</p-calendar>

You can see here how is rendering 2 calendars at the same time because when you change de focus the last calendar doesnt stop rendering instantly. 

![image](https://cloud.githubusercontent.com/assets/23477767/24444728/b3452f0a-1467-11e7-8f4e-ceca4edbe503.png)

![image](https://cloud.githubusercontent.com/assets/23477767/24444851/333cae5e-1468-11e7-8f70-c03ab8ac108e.png)

**What is the motivation / use case for changing the behavior?**
It's a big problem, because with calendar component in the datatable the interaction page is so slow

**Please tell us about your environment:**
SO: Ubuntu 16.04 ls,
IDE: eclipse neon,

* **PrimeNG version:** 2.0.5

* **Browser:** [all | Chrome | Firefox  -->

**Comment by juanravg** [comment]
I have the same problem with Calendar component.

**Comment by igogu** [comment]
It's fixed with angular 4 and primeng 4.0.0 rc2

**Comment by cagataycivici** [comment]
This should be fixed in recent PrimeNG release, please try with it and if the issue persists drop a comment with along with a test case based on plunkr below and we'll review again.

http://plnkr.co/edit/Wj39h1?p=preview
